spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc

```diff
@@ -66,37 +67,214 @@ export MAVEN_PROXY_PORT=8090
+
+[source,bash]
+----
+$ keytool -export -alias dataflow -keystore dataflow.keystore -file dataflow_cert -storepass dataflow
+----
+
+Next, we need to create a truststore which the Shell will use:
+
+[source,bash]
+----
+$ keytool -importcert -keystore dataflow.truststore -alias dataflow -storepass dataflow -file dataflow_cert -noprompt
+----
+
+Now, you are ready to launch the Dataflow Shell using the following JVM arguments:
+
+[source,bash,subs=attributes]
+----
+$ java -Djavax.net.ssl.trustStorePassword=dataflow \
+       -Djavax.net.ssl.trustStore=/path/to/dataflow.truststore \
+       -Djavax.net.ssl.trustStoreType=jks \
+       -jar spring-cloud-dataflow-shell-{project-version}.jar
+----
+
+[TIP]
+====
+In case you run into trouble establishing a connection via SSL, you can enable additional
+logging by using and setting the `javax.net.debug` JVM argument to `ssl`.
+====
+
+Don't forget to target the Dataflow Server with:
+
+[source,bash]
+----
+dataflow:> dataflow config server https://localhost:8443/
+----
+
+[[getting-started-security-enabling-authentication]]
+=== Enabling Authentication
+
+By default, the REST endpoints (administration, management and health), as well
+as Dashboard UI do not require authenticated access. However, authentication can
+be provided via http://oauth.net/2/[OAuth 2.0], thus allowing you to also integrate Spring Cloud
+Data Flow into Single Sign On (SSO) environments. The following 2 OAuth2 Grant Types will be used:
+
+* _Authorization Code_ - Used for the GUI (Browser) integration. You will be redirected to your OAuth Service for authentication
+* _Password_ - Used by the shell (And the REST integration), so you can login using username and password
+
+The REST endpoints are secured via Basic Authentication but will use the Password
+Grand Type under the covers to authenticate with your OAuth2 service.
+
+NOTE: When authentication is set up, it is strongly recommended to enable HTTPS
+as well, especially in production environments.
+
+You can turn on authentication by adding the following to `application.yml` or via
+environment variables:
+
+[source,yaml]
+----
+security:
+  basic:
+    enabled: true                                                     # <1>
+    realm: Spring Cloud Data Flow                                     # <2>
+  oauth2:                                                             # <3>
+    client:
+      client-id: myclient
+      client-secret: mysecret
+      access-token-uri: http://127.0.0.1:9999/oauth/token
+      user-authorization-uri: http://127.0.0.1:9999/oauth/authorize
+    resource:
+      user-info-uri: http://127.0.0.1:9999/me
+----
+
+<1> Must be set to `true` for security to be enabled.
+<2> The realm for Basic authentication
+<3> OAuth Configuration Section
+
+NOTE: As of version 1.0 Spring Cloud Dataflow does not provide finer-grained authorization. Thus, once you are logged in, you have full access to all functionality.
```

Comment:
When using `OAuth`, how can the dataflow shell connect or how can we specify the authentication parameters?